x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -352,6 +353,21 @@ public Query termQuery(Object value, SearchExecutionContext context) {
             throw new IllegalArgumentException(CONTENT_TYPE + " fields do not support term query");
         }
 
+        @Override
+        public Query existsQuery(SearchExecutionContext context) {
+            if (getEmbeddingsField() == null) {
+                return new MatchNoDocsQuery();
+            }
+
+            return NestedQueryBuilder.toQuery(
+                (c -> getEmbeddingsField().fieldType().existsQuery(c)),
+                getChunksFieldName(name()),
+                ScoreMode.None,
+                false,
```

Comment:
I think we want to pull the value for this based on the `index.query.parse.allow_unmapped_fields` index setting. However, we need to make this change for `NestedQueryBuilder` as well.

Can you add a TODO here for this and we can follow up in another PR?

Reply:
Looked into what it would take to properly route the value of the `index.query.parse.allow_unmapped_fields` index setting here. I think we need to add an `allowUnmappedFields` getter to `QueryRewriteContext`, which we can then use here.

We shouldn't do this in this PR, but I don't want to lose track of this tech debt because hard-coding `ignoreUnmapped` to `false` is particularly problematic for exists queries, where it's very valid to successfully return no results when the field does not exist in the mappings. I'll create a follow-up issue to track this.

Reply:
Tracked by https://github.com/elastic/elasticsearch/issues/110156